apps/main/src/dex/features/deposit-gauge-reward/ui/HelperFields.tsx

```diff
@@ -1,18 +1,25 @@
+import { useMemo } from 'react'
 import { useFormContext } from 'react-hook-form'
 import FieldHelperUsdRate from '@/dex/components/FieldHelperUsdRate'
 import { type DepositRewardFormValues } from '@/dex/features/deposit-gauge-reward/types'
-import { useTokensUSDRates } from '@/dex/entities/token'
 import { FlexContainer } from '@ui/styled-containers'
 import { ChainId } from '@/dex/types/main.types'
+import useStore from '@/dex/store/useStore'
 
 export const HelperFields = ({ chainId, poolId }: { chainId: ChainId; poolId: string }) => {
   const { watch } = useFormContext<DepositRewardFormValues>()
   const rewardTokenId = watch('rewardTokenId')
   const amount = watch('amount')
 
-  const {
-    data: [tokenUsdRate],
-  } = useTokensUSDRates([rewardTokenId])
+  const usdRatesMapper = useStore((state) => state.usdRates.usdRatesMapper)
+  const tokens = [rewardTokenId]
+  const tokensKey = JSON.stringify(tokens)
+
+  const [tokenUsdRate] = useMemo(
+    () => tokens.map((token) => (token ? usdRatesMapper[token] : undefined)),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [tokensKey, usdRatesMapper],
+  )
```

Comment:
Why do we need this complicated JSON stringify and ignoring eslint rules?

```suggestion
  const tokenUsdRate = rewardTokenId && usdRatesMapper[rewardTokenId]
```

Reply:
Refactoring this away with a tanstack query in a future PR

Reply:
It's not new, I just inlined whatever `useTokensUSDRates` used to be as this was the only place it was called. As said, will be a tanstack query in the future.